http-client/test-nonet/Network/HTTP/ClientSpec.hs

```diff
@@ -1,17 +1,21 @@
 {-# LANGUAGE OverloadedStrings #-}
 module Network.HTTP.ClientSpec where
 
-import           Control.Concurrent        (forkIO, threadDelay)
```

Comment:
Sorry for the diff in the import section - I'm mostly just keeping things lined up so there are some whitespace changes. The content changes in imports are:

* `import Data.Either (isRight)` added
* `import Network.HTTP.Client.Internal (runProxyOverride)` added -let me know if there is a way that you would prefer to test this functionality without depending on this internal function
* `import System.Environment.Compat (setEnv, lookupEnv, unsetEnv)` in order to replicate the failure that I experienced from inside the test suite. Again, let me know if there is a way that you'd prefer I test this functionality